src/SensitiveFloats.jl

```diff
@@ -154,76 +128,61 @@ each evaluated at (sf1, sf2).
 The result is stored in sf_result.  The order is done in such a way that
 it can overwrite sf1 or sf2 and still be accurate.
 """
-function combine_sfs!{ParamType <: ParamSet,
-                      T1 <: Number, T2 <: Number, T3 <: Number}(
-        sf1::SensitiveFloat{ParamType, T1},
-        sf2::SensitiveFloat{ParamType, T1},
-        sf_result::SensitiveFloat{ParamType, T1},
-        v::T1, g_d::Vector{T2}, g_h::Matrix{T3},
-        calculate_hessian::Bool=true)
+function combine_sfs!{T1 <: Number, T2 <: Number, T3 <: Number}(
+                        sf1::SensitiveFloat{T1},
+                        sf2::SensitiveFloat{T1},
+                        sf_result::SensitiveFloat{T1},
+                        v::T1,
+                        g_d::Vector{T2},
+                        g_h::Matrix{T3})
     # You have to do this in the right order to not overwrite needed terms.
-    if calculate_hessian
+    if sf_result.has_hessian
         combine_sfs_hessian!(sf1, sf2, sf_result, g_d, g_h)
```

Comment:
The reason it was called ```calculate_derivs``` not ```calculate_gradient``` was because of the many places like this in the code where it is not possible to calculate a hessian without having gradients.

You might worry that if the user specifies ```calculate_hessian=true``` but ```calculate_gradient=false```, perhaps expecting to get Hessians but not gradients, strange things might happen.  I'm not sure I'll be able to catch all those spots.  I guess I'll leave it up to your judgement that the new nomenclature is actually clearer.  It wasn't trivial to make sure that Hessians were always zero when ```calculate_derivs``` was false, and I'll trust that you looked carefully to make sure that sensible behavior happens when the user specifies ```calculate_gradients=false``` but ```calculate_hessians=true```.  Better yet would be to just disallow that behavior when you create an SF.